### PR TITLE
transfer GTFS and transit logic

### PR DESCRIPTION
**Pull request by xprogramari**
from diagonal/ repo

### CODE REVIEW

File: src/diagonal.works/b6/ingest/gtfs/gtfs.go [hunk 1]
```diff
@@ -0,0 +1,294 @@
+package gtfs
```
Comment (xprogramari) [inline src/diagonal.works/b6/ingest/gtfs/gtfs.go:1]: i removed timeweights logic from this file / and relying on graph.transittimeweights / graph/tests

File: src/diagonal.works/b6/ingest/gtfs/transit/conflate.go [hunk 2]
```diff
@@ -0,0 +1,259 @@
+package transit
```
Comment (xprogramari) [inline src/diagonal.works/b6/ingest/gtfs/transit/conflate.go:1]: also originally transit logic was in its own dir/ but i figure let's keep it all under ingest/gtfs ?

**Comment by xprogramari** [comment]
(oh sorry wait i forgot to move the binary)

File: src/diagonal.works/b6/cmd/b6-ingest-gtfs/.gitignore
Comment (xprogramari) [inline src/diagonal.works/b6/cmd/b6-ingest-gtfs/.gitignore:1]: do i need this ?? idts / it just happened somehow

File: src/diagonal.works/b6/cmd/b6-ingest-gtfs/b6-ingest-gtfs.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	"log"
+	"runtime"
+
+	"diagonal.works/b6/ingest/compact"
+	"diagonal.works/b6/ingest/gtfs"
+
+	_ "github.com/apache/beam/sdks/go/pkg/beam/io/filesystem/local"
+)
+
+// TODO: add monitoring
```
Comment (xprogramari) [inline src/diagonal.works/b6/cmd/b6-ingest-gtfs/b6-ingest-gtfs.go:14]: we can do that in a separate cl + add monitoring for all binaries ?

File: src/diagonal.works/b6/ingest/gtfs/transit/conflate.go (hunk 2 above)
Comment (xprogramari) [reply src/diagonal.works/b6/ingest/gtfs/transit/conflate.go:1]: HA now i see that's a binary in diagonal, so, should i keep it as a separate thing / do ingest binary for it as well

Comment (xprogramari) [reply src/diagonal.works/b6/ingest/gtfs/transit/conflate.go]: ok i know what the right thing to do it !! added transit binary / moved back transit to its own dir

i tried running just to make sure things work, and gtfs is fine / but transit i guess requires an index w a specific mapping ? if i give it granary square pbf for world i get

Failed to find OSM relation relation/openstreetmap.org/relation/8194082 for trip 214:1

Comment (xprogramari) [reply src/diagonal.works/b6/ingest/gtfs/transit/conflate.go]: ok nvm idk what i m doing !! what do i do w data, there's stuff in data/transit and data/tests/tfl-bus in diagonal repo, do i want both ?

File: src/diagonal.works/b6/cmd/b6-ingest-gtfs/b6-ingest-gtfs.go (hunk 3 above)
Comment (andreweland) [reply src/diagonal.works/b6/cmd/b6-ingest-gtfs/b6-ingest-gtfs.go:14]: Yeah, that seems fine.

File: src/diagonal.works/b6/ingest/gtfs/transit/conflate.go (hunk 2 above)
Comment (andreweland) [reply src/diagonal.works/b6/ingest/gtfs/transit/conflate.go]: You can delete data/tests/tfl-bus and transit/tfl/*.go, because they parse a pre-GTFS feed for TfL busses that I think is no longer supported, so no need to keep it.
The conflation code would be good to keep - it attempts to project a transit route onto the street network, and the heuristics took some time to get right (that's what the evaluate binary is for - to iterate on those heuristics). It shouldn't be coupled to the tfl-bus code. If it is, we can rethink.

Comment (xprogramari) [reply src/diagonal.works/b6/ingest/gtfs/transit/conflate.go]: ok, so transit code is not dependent on the tfl stuff, but transit binary is / in the sense that it tries to conflate / evaluate on gtfs + tfl feed (i.e. multiple networks)
https://github.com/diagonalworks/diagonal/blob/1381c9b6fa109499787232582cfd130a6aeebd2f/src/diagonal.works/diagonal/cmd/transit/transit.go#L17

so delete tfl/ data + code but keep all other the transit/ code is cool 

but

1. should i delete the transit binary + then maybe it does make sense to keep transit logic under ingest-gtfs/ dir or
2. keep the transit binary, assume conflate can be done on multiple gtfs feeds ? and rewrite the binary like that ?

Comment (andreweland) [reply src/diagonal.works/b6/ingest/gtfs/transit/conflate.go]: I'd just delete the transit binary. I think if we were to use it again, it would need some work to properly adapt it to all of the recent changes.

Comment (xprogramari) [reply src/diagonal.works/b6/ingest/gtfs/transit/conflate.go]: chill

File: src/diagonal.works/b6/cmd/b6-ingest-gtfs/.gitignore
Comment (xprogramari) [reply src/diagonal.works/b6/cmd/b6-ingest-gtfs/.gitignore:1]: u re not saying anything so i m just gonna go for it and deal w it late if it s a problem
